src/compiler/scala/tools/nsc/transform/Erasure.scala

```diff
@@ -769,6 +771,14 @@ abstract class Erasure extends InfoTransform
           var qual1 = typedQualifier(qual)
           if (!(qual1.tpe <:< erasure)) qual1 = cast(qual1, erasure)
           Select(qual1, name) copyAttrs tree
+
+        case Apply(fun @ Select(qual, _), args) if fun.symbol.hasAnnotation(definitions.inheritSignatureClass) =>
+          if (!context.enclMethod.owner.isBridge) {
```

Comment:
When's this false?

Reply:
The bridge method forwards to the implementation method. This call has to stay as-is. All other calls to the implementation method are rewritten to target the overridden method.